src/libmeasurement_kit/nettests/runnable.cpp

```diff
@@ -238,74 +238,131 @@ std::string Runnable::generate_output_filepath() {
     return filename.str();
 }
 
+void Runnable::contact_bouncer(Callback<Error> cb) {
+    if (!needs_bouncer) {
+        cb(NoError());
+        return;
+    }
+    auto bouncer = options.get(
+        "bouncer_base_url", std::string{"https://a.collector.ooni.io/bouncer"});
+    logger->info("Contacting bouncer: %s", bouncer.c_str());
+    ooni::bouncer::post_net_tests(
+        bouncer, test_name, test_version, required_test_helpers,
+        [=](Error error, Var<BouncerReply> reply) {
+            if (error) {
+                cb(error);
+                return;
+            }
+            assert(!!reply);
+            auto maybe_collector = reply->get_collector_alternate("https");
+            if (!maybe_collector) {
+                cb(maybe_collector.as_error());
```

Comment:
Should we use a **default collector** in case we are not able to find it with the bouncer?